src/plugins/discover/public/application/apps/main/services/use_discover_state.ts

```diff
@@ -191,6 +202,19 @@ export function useDiscoverState({
     [refetch$, searchSessionManager]
   );
 
+  useEffect(() => {
+    if (!initialSavedSearch) {
+      return;
+    }
+    const newAppState = getStateDefaults({
+      config,
+      data,
+      savedSearch: initialSavedSearch,
+    });
+    stateContainer.replaceUrlAppState(newAppState);
+    setState(newAppState);
+  }, [config, data, initialSavedSearch, reset, stateContainer]);
+
```

Comment:
@majagrubic this quick change made columns work correctly (reset app state with settings of the initialSavedSearch). it should be triggered when `initialSavedSearch` is changing. the name currently is misleading, because it always contains the current saved search.